lib/cli/src/js-package-manager/JsPackageManager.ts

```diff
@@ -115,13 +123,14 @@ export abstract class JsPackageManager {
   /**
    * Return an array of strings matching following format: `<package_name>@<package_latest_version>`
    *
-   * @param packageNames
+   * @param packages
    */
-  public getVersionedPackages(...packageNames: string[]): Promise<string[]> {
+  public getVersionedPackages(...packages: string[]): Promise<string[]> {
     return Promise.all(
-      packageNames.map(
-        async (packageName) => `${packageName}@${await this.getVersion(packageName)}`
-      )
+      packages.map(async (pkg) => {
+        const [packageName, packageVersion] = getPackageDetails(pkg);
+        return `${packageName}@${await this.getVersion(packageName, packageVersion)}`;
```

Comment:
AFAICT this will fail when `packageVersion` is a literal version. `getVersion` calls `satisfies` and if it fails it uses the `latest` version. Consider the case where latest is 1.3.5 and we need version 1.2.0.

Reply:
I think this only fails if the version doesn't exist. I tested `semver.satisfies('1.0.0', '1.0.0')` and it returns `true` so as long as the literal version exists inside `versions.reverse()`, it should find one.

That being said, the `latestVersion` method could probably be hardened against a constraint not being satisfied by all versions on npm. Do you want me to do that as part of this PR?

Reply:
(`current` is only ever set for `@storybook` packages, so the `versionToUse` ternary only ever gets applied to those)

Reply:
Aha my bad